File: backend/src/main/java/com/codeloom/backend/controller/SubmissionController.java
package com.codeloom.backend.controller;

import com.codeloom.backend.dto.SendSubmissionRequest;
import com.codeloom.backend.model.Submission;
import com.codeloom.backend.service.SubmissionService;
import jakarta.validation.Valid;
import java.util.Collection;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/v1/submissions")
@RequiredArgsConstructor
public class SubmissionController {
    private final SubmissionService service;

    @GetMapping
    public Collection<Submission> findSubmissions(Authentication authentication, @RequestParam long problemId) {
        return service.findSubmissions(problemId, authentication);
    }

    @PostMapping
    public void sendSubmission(Authentication authentication, @Valid @RequestBody SendSubmissionRequest request) {
        service.sendSubmission(request, authentication);
    }
}

File: backend/src/main/kotlin/com/codeloom/backend/controller/SubmissionController.kt
package com.codeloom.backend.controller

import com.codeloom.backend.config.BAD_REQUEST_RESPONSE_REF
import com.codeloom.backend.dto.SendSubmissionRequest
import com.codeloom.backend.model.Submission
import com.codeloom.backend.service.SubmissionService
import io.swagger.v3.oas.annotations.responses.ApiResponse
import jakarta.validation.Valid
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/v1/submissions")
class SubmissionController(
    private val submissionService: SubmissionService,
) {
    @GetMapping
    fun findSubmissions(
        auth: Authentication,
        @RequestParam("problemId") problemId: Long,
    ): Collection<Submission> {
        return submissionService.findSubmissions(problemId, auth)
    }

    @PostMapping
    @ApiResponse(responseCode = "400", ref = BAD_REQUEST_RESPONSE_REF)
    fun sendSubmission(
        auth: Authentication,
        @Valid @RequestBody request: SendSubmissionRequest,
    ) {
        submissionService.sendSubmission(request, auth)
    }
}
